lib/ObjWriter/objwriter.cpp

```diff
@@ -713,3 +719,27 @@ void ObjectWriter::EmitDebugModuleInfo() {
     OutContext->setGenDwarfForAssembly(true);
   }
 }
+
+unsigned ObjectWriter::GetEnumTypeIndex(EnumTypeDescriptor TypeDescriptor,
+                                        EnumRecordTypeDescriptor *TypeRecords) {
+  if (ObjFileInfo->getObjectFileType() == ObjFileInfo->IsCOFF)
+    return TypeBuilder.GetEnumTypeIndex(TypeDescriptor, TypeRecords);
+  else
+    return 0;
```

Comment:
We should just not call this at all on Unix and save the p/invoke overhead (ie, this should be `assert(ObjFileInfo->getObjectFileType() == ObjFileInfo->IsCOFF);`